docs/guides/deployment/heroku.rst

```diff
@@ -0,0 +1,99 @@
+.. _ref_guide_deployment_heroku:
+
+=========
+On Heroku
+=========
+
+In this guide we show how to deploy EdgeDB to Heroku using a Heroku PostgreSQL
+add-on as the backend.
+
+Because of Heroku's architecture EdgeDB must be deployed with a web app on
+Heroku. For this guide we will use a `todo app written in node <todo-repo_>`_.
+
+.. _todo-repo: https://github.com/edgedb/simpletodo/tree/main
+
+First copy the code, initialize a new git repo, and create a new heroku app.
+
+.. code-block:: bash
+
+   $ npx tiged 'edgedb/simpletodo#main' simpletodo-heroku
+   $ cd simpletodo-heroku
+   $ git init --initial-branch main
+   $ heroku apps:create --buildpack heroku/nodejs
+
+If you are using the `JS query builder for EdgeDB <js-query-builder>`_ then you
+will need to check the ``dbschema/edgeql-js`` directory in to your git repo
+after running ``yarn edgeql-js``. The ``edgeql-js`` command can not be run
+during the build step on Heroku because it needs access to a running EdgeDB
+instance which is not available at build time on Heroku.
+
+.. _js-query-builder: http://localhost:3000/docs/clients/01_js/index
+
+.. code-block:: bash
+
+   $ yarn install && yarn edgeql-js --target cjs
+
+The ``dbschema/edgeql-js`` directory was added to the ``.gitignore`` in the
+upstream project so we'll remove it here.
+
+.. code-blocK:: bash
+
+   $ sed -i '/^dbschema\/edgeql-js$/d' .gitignore
+
+
+Create a PostgreSQL Add-on
+==========================
+
+Heroku's smallest PostgreSQL plan, Hobby Dev, is too restrictive to run EdgeDB.
+We'll use the Standard 0 plan for this guide.
+
+.. code-block:: bash
+
+   $ heroku addons:create heroku-postgresql:standard-0
+
+
+Add the EdgeDB Buildpack
+========================
+
+To run EdgeDB on Heroku we'll add the `EdgeDB buildpack <buildpack_>`_.
+
+.. _buildpack: https://github.com/edgedb/heroku-buildpack-edgedb
+
+.. code-block:: bash
+
+   $ heroku buildpacks:add \
+       --index 1 \
+       https://github.com/edgedb/heroku-buildpack-edgedb.git
+
+
+Use ``start-edgedb`` in the Procfile
+====================================
+
+To make EdgeDB available to a process prepend the command with ``start-edgedb``
+which is provided by the EdgeDB buildpack. For the sample application in this
+guide, the web process is started with the command ``npm start``. If you have
+other processes in your application besides/instead of web that need to access
+EdgeDB those process commands should be prepended with ``start-edgedb`` too.
+
+.. code-block:: bash
+
+   $ echo "web: start-edgedb npm start" > Procfile
+
+
+Deploy the App
+==============
+
+Commit the changes and push to Heroku to deploy the app.
+
+.. code-block:: bash
+
+   $ git add .
+   $ git commit -m "first commit"
+   $ git push heroku main
+
+It is likely that EdgeDB uses more memory than the default dyno has available.
```

Comment:
Why not just start with `standard-2x`?

Reply:
I didn't find a way to change the dyno without first deploying something. Do you know of a way to do that?

Reply:
Isn't is just changing the earlier step to  `heroku addons:create heroku-postgresql:standard-2`? https://devcenter.heroku.com/articles/heroku-postgres-plans#hobby-tier

Reply:
The memory is needed for EdgeDB server not postgres. On Heroku EdgeDB runs in the same "container" as the web server. The web server needs to be scaled up but I don't think that can be done before there is a web server to scale.

Reply:
What is the default dyno ram size?

Reply:
512MB. Standard 2 is the cheapest one with 1GB.

Reply:
And you've tested that 512 isn't enough experimentally?  What about the master branch with @fantix' pool scaling?

Reply:
> And you've tested that 512 isn't enough experimentally?

Yea it uses more than 700MB

> What about the master branch with @fantix' pool scaling?

I just tried this and it brings memory usage down to a bit more than 400MB which is within the Quota.

Reply:
The app still runs with memory over the limit. Maybe it just takes a performance hit because of swapping?

Reply:
Which memory metric are you looking at?  Adding up all RSS values is not correct, because the processes share a bunch of RAM between them.

Reply:
I'm looking at the memory metrics in the Heroku dashboard

# edgedb-1.1
![image](https://user-images.githubusercontent.com/8278635/158228009-2277dbb0-d33a-4a5a-88a8-31d2692f3505.png)

# edgedb-nightly (with on demand compiler pooling)
![image](https://user-images.githubusercontent.com/8278635/158228603-5acb8926-a169-43fd-97fd-4f7dc5c42960.png)

Reply:
The on-demand version looks reasonable.